src/output.h

```diff
@@ -619,6 +619,24 @@ class scrollingcombattext
 
 extern scrollingcombattext SCT;
 
+std::string wildcard_trim_rule( const std::string &sPatternIn );
+bool wildcard_match( const std::string &sTextIn, const std::string &sPatternIn );
+std::vector<std::string> &wildcard_split( const std::string &s, char delim, std::vector<std::string> &elems );
+template<typename charT>
+int ci_find_substr( const charT &str1, const charT &str2, const std::locale &loc = std::locale() );
+// templated version of ci_equal so it could work with both char and wchar_t
+template<typename charT>
+struct ci_equal {
```

Comment:
We really don't need the generic templated version of that function. `charT` will never be anything but `char`. Same with the function above.

In fact, this will not currently even work with anything else but `char`. Try to call `ci_find_substr<int>(...)` (with suitable dummy arguments) from a source file other than output.cpp. The linker will complain that can not find the function `ci_find_substr<int>`: only the function `ci_find_substr<char>` is (implicitly) instantiated in output.cpp (as it's called from another functions).